mesonbuild/compilers/detect.py

```diff
@@ -1221,9 +1221,17 @@ def detect_swift_compiler(env: 'Environment', for_machine: MachineChoice) -> Com
 
 def detect_nasm_compiler(env: 'Environment', for_machine: MachineChoice) -> Compiler:
     from .asm import NasmCompiler, YasmCompiler, MetrowerksAsmCompilerARM, MetrowerksAsmCompilerEmbeddedPowerPC
-    compilers, _, _ = _get_compilers(env, 'nasm', for_machine)
     is_cross = env.is_cross_build(for_machine)
 
+    # When cross compiling and nasm is not defined in the cross file we can
+    # fallback to the build machine nasm.
```

Comment:
When cross-compiling, I think it makes sense to pick it up from the environment (`PATH` or `NASM`) but not from the native file. That would be surprising.